{
    "source": "BDSA",
    "name": "BDSA-2020-0469",
    "title": "Pydio Vulnerable to Object Injection via Deserialization of Untrusted User Input in 'HttpDownload.php' File",
    "description": "Pydio is vulnerable to object injection via deserialization of untrusted user input. A remote authenticated attacker can exploit this flaw to execute arbitrary code.",
    "technicalDescription": "The `switchAction()` and `detectDLParts()` functions in `core/src/plugins/uploader.http/HttpDownload.php` use an unsafe method of deserialization (the `unserialize()` function), resulting in untrusted input being used without validation. This could allow an authenticated attacker to craft a malicious PHP object that could result in code execution.",
    "publishedDate": "2020-03-18T16:11:21.282Z",
    "updatedDate": "2020-03-23T08:59:58.003Z",
    "disclosureDate": "2020-03-16T00:00:00.000Z",
    "solution": "Fixed in version[** 8.2.4**](https://github.com/pydio/pydio-core/releases/tag/pydio-core-8.2.4) by [this](https://github.com/pydio/pydio-core/commit/f5db57fd859248026a63aaad8edc6939deddc898) commit.\n\nThe latest stable releases are available [here](https://github.com/pydio/pydio-core/releases).",
    "severity": "HIGH",
    "cvss2": {
        "baseScore": 6.5,
        "impactSubscore": 6.4,
        "exploitabilitySubscore": 8.0,
        "severity": "MEDIUM",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "SINGLE",
        "confidentialityImpact": "PARTIAL",
        "integrityImpact": "PARTIAL",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 4.8
        },
        "vector": "(AV:N/AC:L/Au:S/C:P/I:P/A:P/E:U/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 8.8,
        "impactSubscore": 5.9,
        "exploitabilitySubscore": 2.8,
        "severity": "HIGH",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "HIGH",
        "integrityImpact": "HIGH",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "LOW",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 7.7
        },
        "vector": "CVSS:3.0/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H/E:U/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "credit": "Sammy FORGIT - Certilience",
    "workaround": "",
    "vendorFixDate": "2019-11-26T00:00:00.000Z",
    "discoveryDate": "2019-08-08T00:00:00.000Z",
    "vendorNotifiedDate": "2019-08-09T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 7.7,
    "_meta": {
        "allow": [
            "GET"
        ],
        "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-0469",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro2.app.blackduck.com/api/cwes/CWE-502"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/CVE-2019-20453",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-0469/ranges"
            },
            {
                "rel": "reference",
                "href": "https://github.com/pydio/pydio-core/commit/3d577b9cac2758aba52002711cf3fb4b8fe11ac4",
                "label": "INTRODUCTORY_COMMIT"
            },
            {
                "rel": "reference",
                "href": "https://github.com/pydio/pydio-core/commit/f5db57fd859248026a63aaad8edc6939deddc898",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/pydio/pydio-core/releases",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://github.com/pydio/pydio-core/releases/tag/pydio-core-8.2.4",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://www.certilience.fr/2020/03/cve-2019-20452-vulnerabilite-php-object-injection-pydio-core/",
                "label": "ADVISORY"
            },
            {
                "rel": "default-remediation-status",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-0469/default-remediation-status"
            }
        ]
    }
}